From 4e4080233a2843d405a8c0574209a8245baa7de5 Mon Sep 17 00:00:00 2001
From: Alessandro Rubini <rubini@gnudd.com>
Date: Sat, 20 Nov 2010 13:15:48 +0100
Subject: [PATCH] initramfs: stop after one cpio archive

---
 init/initramfs.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/init/initramfs.c b/init/initramfs.c
index 4b9c202..9b446ff 100644
--- a/init/initramfs.c
+++ b/init/initramfs.c
@@ -461,6 +461,7 @@ static char * __init unpack_to_rootfs(char *buf, unsigned len)
 			error("junk in compressed archive");
 		if (state != Reset)
 			error("junk in compressed archive");
+		break; /* so we can use a bigger initrd size in the cmdline */
 		this_header = saved_offset + my_inptr;
 		buf += my_inptr;
 		len -= my_inptr;
-- 
1.5.6.5

