From 2f5ef2e2ea8caf5b69c4acc93c434fd53b5da70d Mon Sep 17 00:00:00 2001
From: Alessandro Rubini <rubini@gnudd.com>
Date: Tue, 6 Mar 2012 01:26:28 +0100
Subject: [PATCH 3/8] nand wrs: our nand is 16-bit connected: fix accordingly

Signed-off-by: Alessandro Rubini <rubini@gnudd.com>
---
 arch/arm/boards/pm9g45/init.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boards/pm9g45/init.c b/arch/arm/boards/pm9g45/init.c
index 629a1ab..e2286ed 100644
--- a/arch/arm/boards/pm9g45/init.c
+++ b/arch/arm/boards/pm9g45/init.c
@@ -47,9 +47,9 @@ static struct atmel_nand_data nand_pdata = {
 	.ale		= 21,
 	.cle		= 22,
 	.det_pin	= -EINVAL,
-	.rdy_pin	= AT91_PIN_PD3,
+	.rdy_pin	= AT91_PIN_PC8,
 	.enable_pin	= AT91_PIN_PC14,
-	.bus_width_16	= 0,
+	.bus_width_16	= 1,
 	.on_flash_bbt	= 1,
 };
 
@@ -73,7 +73,7 @@ static struct sam9_smc_config pm_nand_smc_config = {
 
 static void pm_add_device_nand(void)
 {
-	pm_nand_smc_config.mode |= AT91_SMC_DBW_8;
+	pm_nand_smc_config.mode |= AT91_SMC_DBW_16;
 
 	/* configure chip-select 3 (NAND) */
 	sam9_smc_configure(0, 3, &pm_nand_smc_config);
-- 
1.7.7.2

