diff --git a/kernel/irq/handle.c b/kernel/irq/handle.c
index 27e5c69..3b514a3 100644
--- a/kernel/irq/handle.c
+++ b/kernel/irq/handle.c
@@ -42,6 +42,7 @@ void handle_bad_irq(unsigned int irq, struct irq_desc *desc)
 	kstat_incr_irqs_this_cpu(irq, desc);
 	ack_bad_irq(irq);
 }
+EXPORT_SYMBOL_GPL(handle_bad_irq); /* Temporary: needed for WRVIC */
 
 #if defined(CONFIG_SMP) && defined(CONFIG_GENERIC_HARDIRQS)
 static void __init init_irq_default_affinity(void)

